### PR TITLE
Auto-detect prometheus installation

### PR DESCRIPTION
**Pull request by jakolehm**

### CODE REVIEW

**Comment by nevalla** [comment]
When resolving prometheus service we have to return also prometheus provider id. `MetricsRoute` is using that to fetch metrics with correct queries. I think we are missing that part atm.

**Comment by jakolehm** [comment]
> When resolving prometheus service we have to return also prometheus provider id. `MetricsRoute` is using that to fetch metrics with correct queries. I think we are missing that part atm.

Already pushed a fix for this.

File: src/main/prometheus/operator.ts [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
+      if (!service) return
+
+      return {
+        id: this.id,
+        namespace: service.metadata.namespace,
+        service: service.metadata.name,
+        port: service.spec.ports[0].port
+      }
+    } catch(error) {
+      console.error(error)
```
Comment (nevalla) [inline src/main/prometheus/operator.ts]: maybe this is redundant here?

File: src/main/cluster.ts [hunk 2]
```diff
@@ -95,7 +95,6 @@ export class Cluster implements ClusterInfo {
     this.contextName = kc.currentContext
     this.url = this.contextHandler.url
     this.apiUrl = kc.getCurrentCluster().server
-    await this.contextHandler.init()
```
Comment (nevalla) [inline src/main/cluster.ts:98]: when removing this from `init`, it does not need to be `async` anymore.

**Review by nevalla — APPROVED** [review]
Added couple of minor comments, otherwise LGTM

**Pull request merged**